src/main/python/mainapp.py

```diff
@@ -1,22 +1,33 @@
 from fbs_runtime.application_context.PyQt5 import ApplicationContext
+
+from PyQt5.QtCore import QObject, pyqtSignal
+
 from app import App
+from api import Api
 from login import Login
-from PyQt5.QtCore import QObject, pyqtSignal
 
 
 class MainApp(QObject):
     login_signal = pyqtSignal()
     logout_signal = pyqtSignal()
 
-    def __init__(self, cxt: ApplicationContext, dont_boot_for_test=True, *args, **kwargs):
+    def __init__(
+        self, cxt: ApplicationContext, dont_boot_for_test=True, *args, **kwargs
+    ):
         super(QObject, self).__init__(*args, **kwargs)
         self.dont_boot_for_test = dont_boot_for_test
         self.login = None
         self.app = None
 
         self.cxt = cxt
 
-        self.show_login()
+        with Api("/auth/refresh") as api:
+            status, res = api.post()
+            if status == 200:
```

Comment:
could make this ternary?

Reply:
I decided not to make this ternary for a couple reasons:
* In case we need to add more code based around actions to be taken on either showing the login page or the application
* I think it's a good visual branch in the code to denote showing the login window _or_ the application window.

I'm not opposed to making it ternary but I think I'd prefer to keep it the way it is right now.

Reply:
good points -no ternary here